Update Packed Row documentation (#21132)
yugabyte/yugabyte-db: docs/content/preview/architecture/docdb/persistence.md

## Packed row format

> Packed row format for the YSQL API is [Generally Available](/preview/releases/versioning/#feature-availability), while the support for Packed Row feature in the YCQL API works is still in Tech Preview (/preview/releases/versioning/#feature-availability).

Author: Done.

> Packed row format for the YSQL API is [Generally Available](/preview/releases/versioning/#feature-availability) as of 2.20.0 release, while the support for Packed Row feature in the YCQL API is still in Tech Preview (/preview/releases/versioning/#feature-availability).

Reviewer: @ddhodge - do we prefer GA or Generally Available here?

Author: Dwight responded.

> Packed row format for the YSQL API is [Generally Available](/preview/releases/versioning/#feature-availability) as of 2.20.0 release, while the support for Packed Row feature in the YCQL API is still in Tech Preview (/preview/releases/versioning/#feature-availability).

Reviewer 2: ```suggestion
Packed row format for the YSQL API is {{<badge/ga>}} as of v2.20.0; packed row format for the YCQL API is {{<badge/tp>}}.
```

* Faster multi-column reads, as the reads need to fetch fewer key value pairs.
* UDTs require application rewrite, and therefore not necessarily an option for all use cases, like latency sensitive update workloads.

> The packed row format can be controlled using the [Packed row flags](../../../reference/configuration/yb-tserver/#packed-row-flags).

Reviewer 2: ```suggestion
You configure packed row format using the [Packed row flags](../../../reference/configuration/yb-tserver/#packed-row-flags).
```

### Limitations

> The packed row feature for the YSQL API works across all cross features like backup and restore, schema changes, and so on, subject to certain known limitations which are currently under development:

Reviewer 2: ```suggestion
The packed row feature for the YSQL API works across all cross features, including backup and restore, schema changes, and so on, subject to the following known limitations which are currently under development:
```

> * [#20638](https://github.com/yugabyte/yugabyte-db/issues/20638) Colocated and Packed Row - There is an issue with aggressive garbage collection of the schema versions that are stored in DocDB, in order to interpret Packed Row data. This issue is limited to the colocated table setting, and manifests in certain flavors of compactions. Since it results in non-recoverable errors for colocated workloads, the gflag `ysql_enable_packed_row_for_colocated_table` can be set to false, to avoid the issue on 2.20.1 builds, while the team works to address the issue.

Reviewer 2: ```suggestion
* [#20638](https://github.com/yugabyte/yugabyte-db/issues/20638) Colocated and Packed Row - There is an issue with aggressive garbage collection of the schema versions that are stored in DocDB, in order to interpret Packed Row data. This issue is limited to the colocated table setting, and manifests in certain flavors of compactions. Because this results in non-recoverable errors for colocated workloads, you can set the `ysql_enable_packed_row_for_colocated_table` flag to false, to avoid the issue in v2.20.1.
```

> * [#20638](https://github.com/yugabyte/yugabyte-db/issues/20638) Colocated and Packed Row - There is an issue with aggressive garbage collection of the schema versions that are stored in DocDB, in order to interpret Packed Row data. This issue is limited to the colocated table setting, and manifests in certain flavors of compactions. Since it results in non-recoverable errors for colocated workloads, the gflag `ysql_enable_packed_row_for_colocated_table` can be set to false, to avoid the issue on 2.20.1 builds, while the team works to address the issue.

Reviewer 2: "avoid the issue on 2.20.1 builds" - unclear whether this means the flag is available only in 2.20.1 (or 2.20.1 and later?), or if this is the only way to avoid the issue in 2.20.1.

> * [#21131](https://github.com/yugabyte/yugabyte-db/issues/21131) As of 2.20, new universes created in 2.20+ version will get Packed Row for YSQL is enabled by default. However, if a universe that is on a version lower than 2.20, is upgraded to 2.20, the Packed Row feature for YSQL API will not be enabled. This is because of a known limitation with XCluster universes. The target of XCluster setup might not be able to interpret the Packed Row unless it is upgraded first etc. The team is hardening this limitation as part of this issue.

Reviewer 2: ```suggestion
* [#21131](https://github.com/yugabyte/yugabyte-db/issues/21131) Packed row is enabled by default for YSQL in universes created in v2.20.0 and later. However, packed row is not enabled on universes that are upgraded to v2.20.0 from an earlier version. This is because of a known limitation with xCluster universes, where the target universe in xCluster replication might not be able to interpret the packed row unless it is upgraded before the primary.
```

> * The packed row feature for the YCQL API is still in Tech Preview (/preview/releases/versioning/#feature-availability) in 2.20.0 release. While there are no known limitations, the team is hardeing the feature for YCQL API.

Reviewer 2: ```suggestion
* The packed row feature for the YCQL API is {{<badge/tp>}}. There are no known limitations.
```
